src/hardware/gus.cpp

```diff
@@ -698,33 +716,45 @@ void Gus::RenderUpToNow()
 		last_rendered_ms = now;
 		return;
 	}
-	// Keep rendering until we're current
-	while (last_rendered_ms < now) {
-		last_rendered_ms += ms_per_render;
-		fifo.emplace(RenderFrame());
+
+	const auto elapsed_ms = now - last_rendered_ms;
+	if (elapsed_ms > ms_per_render) {
+		// How many frames have elapsed since we last rendered?
+		const auto num_elapsed_frames = iround(
+		        std::floor(elapsed_ms / ms_per_render));
```

Comment:
Are you sure we aren't swallowing/skipping sample frames sometimes because of the rounding?

Reply:
Yes; this errs on the side of whole frames only for the sub-millisecond chunk of time, because the game or program at a minimum needs to make sure at least that many samples have been DMA'd into the card - if we round up, then we might go beyond what's available.

But to to sure - no matter what, during the audio callback where the mixer pulls frames, it gets all the frames at that point.  So no frames are dropped or lost.